Miscellaneous updates to developer guide

david-eom opened the pull request:
This PR addresses issues assigned pertaining to DG:
* Update `Model` UML diagram accordingly
* Get rid of AB references in `Model` section
* Addition of tracking history use case (UC15)
* Beefing up "Appendix: Instructions for manual testing"

Task remaining

- [x] UML diagram for command history (will do in a separate commit by 7 Nov 1200hrs)

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+1. Enter `add n/aa p/12345678` in the command box.
+
+	Expected: A new contact card with name "aa" and phone number "12345678" is created. The contact card is most likely to be appearing at the top, since SociaLite arranges the contact cards according to their names alphabetically.
```
For the second sentence, maybe we can say "This contact card will likely appear at the top, as SociaLite displays records in alphabetical order of contacts' names."?

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+4. Enter `add n/aa p/87654321` in the command box.
+
+	Expected: A new contact card with name "aa" and phone number "87654321" is created. SociaLite allows repetitive names in the contact list, but uses phone number as the unique identifier for a contact.
```
Maybe can say "SociaLite allows names to be repeated in the contact list as it uniquely identifies contacts through their phone number."

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+5. Enter `add n/John Appleseed p/13572468 fb/john.appleseed twitter/morethanfifteenchars`.
+
+	Expected: SociaLite shows error message, indicating that the telegram username does not follow their naming convention.
```
I think you mean "Twitter handle" instead of "telegram username" here.

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+1. Enter `edit 1 n/abc` in the command box.
+
+	Expected: the first contact card's name changes to "abc". It might not be the first contact card in the list due to alphabetical ordering.
```
For the second sentence, you might wanna say "After this modification, this contact card might no longer be the first in the list due to alphabetical ordering."

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+### Editing a contact
+
+Assuming all test cases in the previous section were carried out, and that the first contact card in the list is "aa", and there are only 10 contacts stored in SociaLite:
```
Maybe we can say "Assuming you have performed all preceding test cases in order, you should have 10 contacts stored in SociaLite, with the contact card 'aa' appearing at the top of the list."

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+4. Enter `edit 1 fb/hello` in the command box.
+
+	Expected: the first contact card in SociaLite should have a Facebook icon showed up at the bottom, with its handle being "hello".
```
Can change to 'Expected: the first contact card in SociaLite should now display a Facebook icon at the bottom, with the handle "hello".'

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+1. Enter `delete 1` in the command box.
+
+	Expected: the first contact card in SociaLite will be deleted. There will be 9 contacts in SociaLite.
```
"There will be 9 contacts left in SociaLite."

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+2. Enter `find a t` in the command box.
+
+	Expected: all contacts whose name starts with "a" and "t" (e.g. Aaron Tan, Alice Teo, Arthur Taylor) will show up in the list.
```
Sorry! I'm just suggesting this change for greater clarity, in case the reader assumes that names like "Tony Ang" and "Terry Au" will be displayed as well.

"Expected: all contacts whose name contain two or more words, with the first word starting with 'a' and the second word starting with 't' (e.g. Aaron Tan, Alice Teo, Arthur Taylor) will show up in the list."

david-eom replied:
Actually for this one I believe the implementation will actually show both "Aaron Tan" and "Tony Ang"! But this is a great point, I will include this in the DG as well!

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+To test out this feature, you can choose to add your own contact information to SociaLite with your own social media handle. For example, you can enter `add n/<your name> p/<your phone number> ig/<your instagram handle>`.
+
+1. Hover your cursor on top of your Instagram handle in SociaLite.
```
Can change to "Hover your cursor over...". (same for line 905)

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+2. Click your mouse when it is hovering on top of a certain handle.
+
+	Expected: SociaLite will open your default web browser and direct you to the specific social media page (e.g. your own Instagram profile).
```
Maybe we can say "SociaLite will open your default web browser and launch your chosen social media page (e.g. your own Instagram profile)."

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+
+	Expected: The sixth contact card in the list will be pinned to the top of the list. The button "Pin" will change colour and its text will become "Unpin".
+
+7. Click on the "Unpin" button of the first contact card in the list with your mouse 
```
Can add a full stop at the end of this line, seeing how you've already added full stops for all the previous steps. 😆

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+7. Click on the "Unpin" button of the first contact card in the list with your mouse 
 
-### Launch and shutdown
+	Expected: same as test case number 2.
```
Maybe it'll be better to elaborate on the expected outcome here because the outcome for Test Case 2 makes reference to the third contact in brackets. 

So we can say 'Expected: same as test case number 2. The chosen contact card will be unpinned and return to its original position in the list (i.e. in alphabetical order of names). The button "Unpin" will return to its original colour and its text will display "Pin".'.

stanley-1 commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
 
-   1. Re-launch the app by double-clicking the jar file.<br>
-       Expected: The most recent window size and location is retained.
+	Expected: Socialite shows that the contact information of the fifth contact has been copied to the system clipboard. The "Share" button will change its text to "Copied!" for two seconds, before reverting back to "Share".
```
Minor nit: Remove 'back' after 'reverting'

stanley-1 reviewed (approved):
Thanks for creating the sequence of test cases @david-eom! Very very comprehensive work done! Thanks for including full stops at the end of all the earlier sentences too haha! Your ModelClassDiagram was well done as well!

david-eom commented on docs/DeveloperGuide.md:
```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+1. Launching
+   1. Double-click on the `jar` file icon in your OS' GUI
+   
+   	Expected: the GUI for SociaLite app shows up with a set of sample contacts.
```
Oops actually not entirely sure about how to do so! (cos it rendered fine on intellij...🥲)

<img width="509" alt="截圖 2021-11-07 下午12 07 44" src="https://user-images.githubusercontent.com/66581587/140631972-1dbeb0b4-598c-4c98-b555-d8337d624cfc.png">

Zacchua replied:
hmm i get the same as stanley when i view on github though. Perhaps it would be better to optimize for github because our testers would be viewing this on github?

stanley-1 replied:
Haha maybe it's just a glitch on GItHub which happens to my docs sometimes too! As long as it's rendered correctly on GitHub pages, it should be fine!

david-eom commented:
Thanks @stanley-1 for the comprehensive feedback! I have addressed most of the issues brought up, but 2 remains: the indentation for "launching and exiting" and the part on find. I was thinking the indentation might be caused by header not being identified, and it _should_ be fixed now... But pls do give it a look!

Zacchua commented on docs/diagrams/CommandHistorySequenceDiagram.puml:
```diff
@@ -0,0 +1,58 @@
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":SociaiteParser" as SocialiteParser LOGIC_COLOR
```
SocialiteParser instead of SociaiteParser

stanley-1 replied:
Typo: Should use `:SocialiteParser` instead.

Zacchua commented on docs/DeveloperGuide.md:
```diff
@@ -208,6 +203,17 @@ Represented below is the sequence diagram when `picture INDEX` is executed. For
+
+Command history allows users to navigate through commands that they have entered previously by pressing `UP/DOWN` key in the command box. It resembles the behavior of a macOS or Windows terminal. As a result, users can simply make slight edit to the previous command to rectify a typo.
+
+Command history from all previous sessions will be stored in `commandhistory.json`, and loaded as an list of strings in `CommandHistory` class when SociaLite is launched. The command history of the current session will be added to the list of strings, and written back to the `.json` file upon exiting SocaiLite.
```
"a list of strings" instead of "an list of strings"

stanley-1 commented on docs/diagrams/CommandHistorySequenceDiagram.puml:
```diff
@@ -0,0 +1,58 @@
+
+LogicManager -> XYZcommand : execute()
+activate XYZcommand
+XYZcommand --> LogicManager : result
```
To be in line with the variable name used in `LogicManager.java`, do you want to change `result` to `commandResult` instead?

david-eom replied:
Yep, I think it should be changed to `commandResult` also! Thanks for pointing that out!

Zacchua reviewed (approved):
Great job beefing up the DG xiaozhi! Just a couple of small nits to fix!

stanley-1 reviewed (approved):
The sequence diagram is clean and easy to interpret! Good job! Just gotta fix a typo!

david-eom commented:
Hi @stanley-1 @Zacchua, in the latest commit f08a7f3 I've tried to fixed the indentation issue, should be working now!
<img width="460" alt="截圖 2021-11-08 上午12 14 10" src="https://user-images.githubusercontent.com/66581587/140652923-f67115aa-97b3-42ca-83ca-1d8e797f8fba.png">
